src/main/java/org/brapi/test/BrAPITestServer/repository/BrAPIRepositoryImpl.java

```diff
@@ -14,36 +26,90 @@
-		searchQuery = applyUserId(searchQuery);
+	/**
+	 * 	Use this method to page simple entities with no lazily loaded collections or attributes.
+	 * 	WARN: Failure to do so can easily exhaust memory at scale and will cause hibernate warnings.
+	 */
+	public Page<T> findAllBySearchAndPaginate(SearchQueryBuilder<T> searchQuery, Pageable pageReq)
+		throws InvalidPagingException {
+		applyUserId(searchQuery);
+
+		Long totalCount = getTotalCount(searchQuery);
+
+		validatePageNumber(pageReq, totalCount);
+
+		if (totalCount == 0) {
+			// Short-circuit to avoid running possible long-running query
+			return new PageImpl<>(Collections.emptyList(), pageReq, totalCount);
+		}
+
 		List<T> content = getPagedContent(searchQuery, pageReq);
+
+        return new PageImpl<>(content, pageReq, totalCount);
+	}
+
+	/**
+	 * This method should be used when there is a need to page entities that are complex and have lots of lazily loaded,
+	 * one-to-many collection attributes. Call this method with the query you've built.
+	 * WARN: To avoid multiple bag fetch hibernate errors, only one collection that is one-many can be fetched at a time.
+	 * Ensure only one of these types of entity members is being fetched at time.
+	 * See {@link org.brapi.test.BrAPITestServer.service.germ.GermplasmService#findGermplasmEntities(GermplasmSearchRequest, Metadata)} for example usage.
+	 *
+	 * Once you fetch the first lazy loaded collection, if there are others you need to fetch for your entity it is
+	 * recommended to utilize the non-paging findAllBySearch() method, creating a searchQuery that inserts all the ids
+	 * of the entities found in the results from this method.  Again, follow code path of above example usage to see how this is done.
+	 *
+	 * If the need is to fetch according to an entirely different base criteria, feel free to call this method again.
+	 *
+	 * The results will be paged and ordered based on the submitted searchQuery and pageReq.
+    */
+	public Page<T> findAllBySearchPaginatingWithFetches(SearchQueryBuilder<T> searchQuery, Pageable pageReq)
+		throws InvalidPagingException {
+		applyUserId(searchQuery);
+
 		Long totalCount = getTotalCount(searchQuery);
 
-		Page<T> page = new PageImpl<>(content, pageReq, totalCount);
+		validatePageNumber(pageReq, totalCount);
+
+		// First grab all the ids of the entities according to the criteria of the searchQuery, paging as specified in the pageReq.
+		List<String> content = getPagedContentIdsOnly(searchQuery, pageReq);
 
-		return page;
+		Page<String> pagedIds = new PageImpl<>(content, pageReq, totalCount);
+
+		// Now execute another query to fetch all the entities requested in the searchQuery, passing the pagedIds found
+		// in the previous query.  We will fetch them utilizing the ids from the paged query, avoiding the hibernate
+		// warning of paging while fetching and consuming considerably less memory.
+        return findAllBySearchUsingIds(searchQuery, pagedIds, pageReq);
+	}
```

Comment:
This looks like it's not indented either, might be a GitHub display issue though.
```suggestion
		// warning of paging while fetching and consuming considerably less memory.
                return findAllBySearchUsingIds(searchQuery, pagedIds, pageReq);
	}
```

Reply:
Yea, this is indented in my IntelliJ:
<img width="648" alt="Screenshot 2025-03-28 at 12 27 58 PM" src="https://github.com/user-attachments/assets/6f8472cb-7b2a-4e09-bc9d-25fd2a531c75" />